### PR TITLE
[feature] Migrate door prefab + anims

### LINKED ISSUE #31: Migrate Door Controller

### LINKED ISSUE #32: Door Open-Close Animation

### PR DESCRIPTION
**Pull request by rafiftaris**
### Main changes
- add door prefab
- add door anim (open/close)

### Tasks List
- [close #31 ]
- [close #32 ]

### Attachments

https://user-images.githubusercontent.com/51603162/149952160-a2db7fce-c349-46e6-b78d-c916d916c484.mp4

### CODE REVIEW

**Review by naufalhilmi12345 — CHANGES_REQUESTED** [review]
player to close to close the door so player move a little when close the door

**Review by naufalhilmi12345 — APPROVED** [review]
Cool! LGTM

File: Assets/Dev/Tests/PlayerTest.cs [hunk 1]
```diff
@@ -145,11 +165,53 @@ public IEnumerator PlayerInventory_ScrollActiveItem()
     public IEnumerator PlayerInteractableDetector_InteractLightSwitch()
     {
         yield return new WaitWhile(() => sceneLoaded == false);
-        inputTestFixture.Press(KeyboardMouseTestFixture.RegisteredInput.Interact);
-        yield return null;
+        inputTestFixture.Press(KeyboardMouseTestFixture.RegisteredInput.MoveRight);
+        float moveDuration = GetMovementDurationTowards(GameObject.Find("LightSwitch").transform);
```
Comment (raihansyahdean) [inline Assets/Dev/Tests/PlayerTest.cs:169]: LightSwitch object is searched more than once. Better save this object to a variable to do less computation

File: Assets/Dev/Tests/PlayerTest.cs [hunk 2]
```diff
@@ -145,11 +165,53 @@ public IEnumerator PlayerInventory_ScrollActiveItem()
...
 
         ILightSwitchController lightSwitchController = GameObject.Find("LightSwitch").GetComponent<ILightSwitchController>();
+
+        inputTestFixture.Press(KeyboardMouseTestFixture.RegisteredInput.Interact);
+        yield return null;
+        inputTestFixture.Release(KeyboardMouseTestFixture.RegisteredInput.Interact);
+        yield return null;
         Assert.IsTrue(lightSwitchController.GetState());
+
+        inputTestFixture.Press(KeyboardMouseTestFixture.RegisteredInput.Interact);
```
Comment (raihansyahdean) [inline Assets/Dev/Tests/PlayerTest.cs:181]: When turning the lights on there are both press and release calls. But when turning the lights off only press is called. Should there also be a Release call for consistency?

File: Assets/Dev/Scripts/Interactables/Controller/DoorController.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ * Inherit Interactable.
+ */
+public class DoorController : Interactable, IDoorController
+{
+    #region Variables
+    private const string animParam = "isOpen";
+    [Header("Door States")]
+    [SerializeField]
+    [Tooltip("True if door is in open state")]
+    private bool isOpen = false;
```
Comment (raihansyahdean) [inline Assets/Dev/Scripts/Interactables/Controller/DoorController.cs:20]: Rename to: _isOpen

File: Assets/Dev/Scripts/Interactables/Controller/DoorController.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+/*
+ * Class to control door states and animation.
+ * Inherit Interactable.
+ */
+public class DoorController : Interactable, IDoorController
+{
+    #region Variables
+    private const string animParam = "isOpen";
```
Comment (raihansyahdean) [inline Assets/Dev/Scripts/Interactables/Controller/DoorController.cs:16]: Rename to : _animParam

File: Assets/Dev/Scripts/Interactables/Controller/DoorController.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    {
+        //Debug.Log("[INTERACTABLE] " + (isOpen ? "Closing " : "Opening ") + this.name);
+        isOpen = !_doorAnim.GetBool(animParam);
+        _doorAnim.SetBool(animParam, isOpen);
+    }
+
+    public override void OnInteraction()
+    {
+        ChangeState();
+        Debug.Log("[INTERACTABLE] Door interacted");
```
Comment (raihansyahdean) [inline Assets/Dev/Scripts/Interactables/Controller/DoorController.cs:49]: Comment this line

File: Assets/Dev/Scripts/Interactables/Controller/DoorController.cs [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    public float GetAngle()
+    {
+        return transform.parent.rotation.y;
+    }
+
+    // General function to change the state of doors
+    private void ChangeState()
+    {
+        //Debug.Log("[INTERACTABLE] " + (isOpen ? "Closing " : "Opening ") + this.name);
+        isOpen = !_doorAnim.GetBool(animParam);
```
Comment (raihansyahdean) [inline Assets/Dev/Scripts/Interactables/Controller/DoorController.cs:42]: Better use isOpen = !isOpen for simplicity

**Review by raihansyahdean — CHANGES_REQUESTED** [review]
- Add interactable icon for doors (use 3D Exclamation Mark prefab for now)
- Move "VoxelObjs" to "Arts" folder

**Review by raihansyahdean — APPROVED** [review]

**Pull request merged**